src/network/MirrorNodeService.js

```diff
@@ -0,0 +1,190 @@
+                    .then(
+                        (
+                            /** @type {import("axios").AxiosResponse<{ tokens: MirrorNodeTokenResponse[] }>} */ response,
+                        ) => {
+                            /** @type {HashgraphProto.proto.ITokenBalance[]} */
+                            const tokenBalances = response.data.tokens.map(
+                                (
+                                    /** @type {MirrorNodeTokenResponse} */ token,
+                                ) => ({
+                                    tokenId: TokenId.fromString(
+                                        token.token_id,
+                                    )._toProtobuf(),
+                                    balance: Long.fromNumber(token.balance),
+                                    decimals: token.decimals,
+                                }),
+                            );
+
+                            resolve(tokenBalances);
+                        },
+                    )
+                    .catch((error) => {
+                        reject(error);
+                    });
+            }, this._timeout);
+        });
+    }
+
+    /**
+     * @param {string} idOrAliasOrEvmAddress
+     * @returns {Promise<HashgraphProto.proto.ITokenRelationship[]>}
+     */
+    getTokenRelationshipsForAccount(idOrAliasOrEvmAddress) {
+        return new Promise((resolve, reject) => {
+            setTimeout(() => {
+                this._mirrorNodeGateway
+                    .getAccountTokens(idOrAliasOrEvmAddress)
+                    .then(
+                        (
+                            /** @type {import("axios").AxiosResponse<{ tokens: MirrorNodeTokenResponse[] }>} */ response,
+                        ) => {
+                            /** @type {HashgraphProto.proto.ITokenRelationship[]} */
+                            const tokenRelationships = response.data.tokens.map(
+                                (
+                                    /** @type {MirrorNodeTokenResponse} */ token,
+                                ) => ({
+                                    tokenId: TokenId.fromString(
+                                        token.token_id,
+                                    )._toProtobuf(),
+                                    balance: Long.fromNumber(token.balance),
+                                    decimals: token.decimals,
+                                    kycStatus: this.getTokenKycStatusFrom(
+                                        token.kyc_status,
+                                    ),
+                                    freezeStatus: this.getTokenFreezeStatusFrom(
+                                        token.freeze_status,
+                                    ),
+                                    automaticAssociation:
+                                        token.automatic_association,
+                                }),
+                            );
```

Comment:
We should not be manually mapping from JSON output to a different JavaScript object format. We should be using the OpenAPI spec to generate these models instead of crafting them manually.

Reply:
At the very beginning the team has discussed multiple approaches for implementation and has chosen the one that's currently implemented. We are ware using OpenAPI is the better approach but as we've discussed in the meeting this is more realistic to happen in SDK V3.